### PR TITLE
Extract ReadsSource from DataflowReadsPipeline into a public class in the engine

### PR DESCRIPTION
**Pull request by droazen**
This class is now needed by other pipelines, so let's make it generally available
in the engine.

### CODE REVIEW

**Comment by droazen** [comment]
For @lbergelson to review please

File: src/main/java/org/broadinstitute/hellbender/engine/dataflow/ReadsSource.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+/**
+ * Class to load reads into a PCollection from either a cloud storage bucket or a local bam file.
+ */
+public class ReadsSource {
+    private final String bam;
+    private final boolean cloudStorageUrl;
+    private GCSOptions options;
+    private GenomicsFactory.OfflineAuth auth;
+
+    public ReadsSource(String bam, File clientSecret){
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/engine/dataflow/ReadsSource.java]: We should switch out strings here for some URI class.  I don't think the existing URI classes support raw paths as a valid input so we probably need to create a wrapper.  We don't want to force people to write `file://somepath` on the command line.

Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/engine/dataflow/ReadsSource.java]: Will create a ticket to write a class that wraps URI and pre-pends a "file://" prefix when there's no prefix

File: src/main/java/org/broadinstitute/hellbender/engine/dataflow/ReadsSource.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        this.bam = bam;
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/engine/dataflow/ReadsSource.java]: We should add defensive null checks here.

**Comment by lbergelson** [comment]
This should have tests for what we can test.  I'll make the changes.

**Comment by lbergelson** [comment]
Adding tests separately since this is blocking @droazen

**Pull request merged**